src/oid.rs

```diff
@@ -193,47 +168,11 @@ impl ObjectId {
         buf
     }
 
-    // Generates a new machine id represented as an MD5-hashed 3-byte-encoded hostname string.
-    // Represented in Little Endian.
-    fn gen_machine_id() -> Result<[u8; 3]> {
-        // Short-circuit if machine id has already been calculated.
-        // Since the generated machine id is not variable, arising race conditions
-        // will have the same MACHINE_BYTES result.
-        unsafe {
-            if let Some(bytes) = MACHINE_BYTES.as_ref() {
-                return Ok(*bytes);
-            }
-        }
-
-        let hostname = get_hostname();
-        if hostname.is_none() {
-            return Err(Error::HostnameError);
-        }
-
-        // Hash hostname string
-        let digest = md5::compute(hostname.unwrap().as_str());
-        let hash = format!("{:x}", digest);
-
-        // Re-convert string to bytes and grab first three
-        let mut bytes = hash.bytes();
-        let mut vec: [u8; 3] = [0; 3];
-        for v in &mut vec {
-            match bytes.next() {
-                Some(b) => *v = b,
-                None => break,
-            }
-        }
-
-        unsafe { MACHINE_BYTES = Some(vec) };
-        Ok(vec)
-    }
-
-    // Gets the process ID and returns it as a 2-byte array.
-    // Represented in Little Endian.
-    fn gen_process_id() -> [u8; 2] {
-        let pid = unsafe { libc::getpid() as u16 };
-        let mut buf: [u8; 2] = [0; 2];
-        LittleEndian::write_u16(&mut buf, pid);
+    // Generate a random 5-byte array.
+    fn gen_process_id() -> [u8; 5] {
+        let rng = thread_rng().gen_range(0, MAX_U24) as u32;
+        let mut buf: [u8; 5] = [0; 5];
+        BigEndian::write_u32(&mut buf, rng);
```

Comment:
Actually @saghm do you know if there are any requirements here as far as Big vs Little endian here? The specifications say timestamp and counter are BigEndian but don't mention the random value, does that imply that this should still be LittleEndian?

Reply:
Realistically, I'm not sure it matters (since the bytes are random, I can't see how the order they're inserted would make a difference), but from a quick check, it appears that Python [uses `">"` for the endianness](https://github.com/mongodb/mongo-python-driver/blob/357e3e3d6840f94794fb50ccf68734f483dae7d6/bson/objectid.py#L45), which from Googling appears to be [big endian](https://docs.python.org/2/library/struct.html#byte-order-size-and-alignment]

Reply:
cool yeah, I figured the randomness makes it moot but wanted to check. Thanks!